server/data/data-api/src/test/java/org/apache/james/rrt/lib/CanSendFromContract.java

```diff
@@ -134,13 +134,22 @@ default void userCanSendFromShouldBeTrueWhenSenderIsAnAliasInAnotherDomainOfAnAl
 
     @Test
     default void userCanSendFromShouldBeTrueWhenSenderIsAnAliasOfTheDomainUser() throws Exception {
-        Username fromUser = USER.withOtherDomain(Optional.of(OTHER_DOMAIN));
+        Username fromUser = USER.withOtherDomain(OTHER_DOMAIN);
 
         redirectDomain(OTHER_DOMAIN).to(DOMAIN);
 
         assertThat(canSendFrom().userCanSendFrom(USER, fromUser)).isTrue();
     }
 
+    @Test
+    default void userCanSendFromShouldBeFalseWhenDomainMapping() throws Exception {
+        Username fromUser = USER.withOtherDomain(OTHER_DOMAIN);
+
+        addDomainMapping(OTHER_DOMAIN, DOMAIN);
```

Comment:
Use `redirect(..).to(..)` syntax

Reply:
Would you agree with `redirect(...).to(...).asMapping()` & `redirect(...).to(...).asAlias()`?

Reply:
sounds nice